gitlab/types.py

```diff
@@ -31,8 +31,43 @@ def set_from_cli(self, cli_value: Any) -> None:
     def get_for_api(self) -> Any:
         return self._value
 
+    def get_as_tuple_list(self, *, key: str) -> List[Tuple[str, Any]]:
+        return [(key, self._value)]
+
+
+class ArrayAttribute(GitlabAttribute):
+    """To support `array` types as documented in
+    https://docs.gitlab.com/ee/api/#array"""
+
+    def set_from_cli(self, cli_value: str) -> None:
+        if not cli_value.strip():
+            self._value = []
+        else:
+            self._value = [item.strip() for item in cli_value.split(",")]
+
+    def get_for_api(self) -> str:
+        # Do not comma-split single value passed as string
+        if isinstance(self._value, str):
+            return self._value
+
+        if TYPE_CHECKING:
+            assert isinstance(self._value, list)
+        return ",".join([str(x) for x in self._value])
+
+    def get_as_tuple_list(self, *, key: str) -> List[Tuple[str, str]]:
+        if isinstance(self._value, str):
+            return [(f"{key}[]", self._value)]
+
+        if TYPE_CHECKING:
+            assert isinstance(self._value, list)
+        return [(f"{key}[]", str(value)) for value in self._value]
+
+
+class CsvStringAttribute(GitlabAttribute):
+    """For values which are sent to the server as a Comma Separated Values
+    (CSV) string.  We allow them to be specified as a list and we convert it
+    into a CSV"""
```

Comment:
nitpick: since we changed `ListAttribute` to `ArrayAttribute`, could we now call this `CommaSeparatedListAttribute` to follow upstream API naming? CSV almost sounds closer to RFC 4180 / https://docs.python.org/3/library/csv.html which I associate more with file specs/parsing.

![image](https://user-images.githubusercontent.com/16777978/145674383-f62650bf-ba6c-47d8-b32f-a14e47922ad8.png)

Reply:
Good idea. Thanks and done!